### PR TITLE
config: x86/64: add back some options from config-6.6

### PR DESCRIPTION
**Pull request by graysky2**
Miscell. options that did not make it from 6.6 to 6.12

### CODE REVIEW

**Comment by namiltd** [comment]
Quite a lot of changes. They should be looked at individually. Maybe not all added settings exist in kernel 6.12.
Please rebase.

**Comment by graysky2** [comment]
Rebased, was meant to only changes to single file.

**Comment by namiltd** [comment]
git reset --hard 20f2f68b4432f1a34d7c8cc5742fcedc38b02a57
git cherry-pick d42b3a058a6f9c2c39255bdfc5f42b8452b9fabd
git push --force

**Comment by namiltd** [comment]
I took a look at the changes made and this is what happened:
CONFIG_AMD_IOMMU_V2 does not exist in kernel 6.12
CONFIG_CONFIG_BLK_DEV_INTEGRITY_T10 does not exist in kernel 6.12
CONFIG_CALL_THUNKS_DEBUG exist as CONFIG_MITIGATION_CALL_THUNKS_DEBUG
You need to review the remaining variables and for the first two mentioned, check if they are not under a different name.

Similarly for https://github.com/namiltd/openwrt/pull/51 only CONFIG_CRASH_HOTPLUG exists in kernel 6.12.

**Comment by namiltd** [comment]
@LynxChaus They don't pass github check, can't be as modules, etc. I'm working on it, I'll enable them again when I find a solution.

**Comment by graysky2** [comment]
> CONFIG_CALL_THUNKS_DEBUG exist as CONFIG_MITIGATION_CALL_THUNKS_DEBUG

How did you determine that the old name changed?  Commit messages?

**Comment by namiltd** [comment]
Strange, I'm looking at the 6.12.1 sources now and CALL_THUNKS_DEBUG still exists in arch/x86/Kconfig

**Comment by namiltd** [comment]
My mistake, I confused CALL_DEPTH_TRACKING with CALL_THUNKS_DEBUG

**Comment by graysky2** [comment]
I have been running with this PR + your latest commit with no issues.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne
Run-tested: x86/64/AMD Cezanne

File: target/linux/x86/64/config-6.12 [hunk 1]
```diff
@@ -80,14 +82,22 @@ CONFIG_BLK_MQ_VIRTIO=y
 CONFIG_BLK_PM=y
 CONFIG_BOOT_VESA_SUPPORT=y
 CONFIG_BTT=y
+CONFIG_CALL_DEPTH_TRACKING=y
+CONFIG_CALL_PADDING=y
+CONFIG_CALL_THUNKS=y
+# CONFIG_MITIGATION_CALL_THUNKS_DEBUG is not set
```
Comment (namiltd) [inline target/linux/x86/64/config-6.12:88]: +# CONFIG_CALL_THUNKS_DEBUG is not set

CONFIG_MITIGATION_CALL_THUNKS_DEBUG does not exist in kernel 6.12 (my mistake)

**Comment by namiltd** [comment]
> I have been running with this PR + your latest commit with no issues.
> 
> Build system: x86/64 Build-tested: x86/64/AMD Cezanne Run-tested: x86/64/AMD Cezanne

Have you found any uses without this PR? I'm not sure if adding these settings is necessary or if it will change anything at all.

**Comment by graysky2** [comment]
I have been running with this PR applied.  Figured there was a reason these were added to 6.6.

**Comment by namiltd** [comment]
Some settings require subsequent ones, so they may be leftovers from previous settings and are no longer required.

**Comment by graysky2** [comment]
I don't know.  Many of these are enabled on my Arch box.
```
ACPI_VIDEO
CONFIG_ACPI_VIDEO=m

AUXILIARY_BUS
CONFIG_AUXILIARY_BUS=y

CALL_DEPTH_TRACKING
CONFIG_MITIGATION_CALL_DEPTH_TRACKING=y

CALL_PADDING
CONFIG_CALL_PADDING=y

CALL_THUNKS
CONFIG_CALL_THUNKS=y

MITIGATION_IBRS_ENTRY
CONFIG_MITIGATION_IBRS_ENTRY=y

MITIGATION_SRSO
CONFIG_MITIGATION_SRSO=y

MITIGATION_UNRET_ENTRY
CONFIG_MITIGATION_UNRET_ENTRY=y

CRYPTO_ECB
CONFIG_CRYPTO_ECB=y

DRM_BUDDY
CONFIG_DRM_BUDDY=m

DRM_DISPLAY_DP_HELPER
CONFIG_DRM_DISPLAY_DP_HELPER=y

DRM_DISPLAY_HDCP_HELPER
CONFIG_DRM_DISPLAY_HDCP_HELPER=y

DRM_DISPLAY_HDMI_HELPER
CONFIG_DRM_DISPLAY_HDMI_HELPER=y

DRM_DISPLAY_HELPER
CONFIG_DRM_DISPLAY_HELPER=m

DRM_MIPI_DSI
CONFIG_DRM_MIPI_DSI=y

VMAP_PFN
CONFIG_VMAP_PFN=y

ZLIB_DEFLATE
CONFIG_ZLIB_DEFLATE=y
```

**Comment by namiltd** [comment]
I suggest not touching any settings for now. Remember that some are enabled by modules and do not need to be in config.

**Pull request closed without merging**